mono/mini/debugger-agent.c

```diff
@@ -10397,4 +10547,71 @@ mono_debugger_agent_parse_options (char *options)
 	sdb_options = options;
 }
 
+// Called from main thread
+static void burst_mono_shutdown()
+{
+	burst_lock();
+	// Leave the domain registered (as this prevents mono from trying to decode any mmethods belonging solely to burst)
+	// But we clear the callbacks, because the callbacks will be torn down in burst shortly
+	g_BurstDebugCallbacks.BurstFinishSetup = NULL;
+	g_BurstDebugCallbacks.BurstIsFileBursted = NULL;
+	g_BurstDebugCallbacks.BurstBreakpointNotify = NULL;
+	g_BurstDebugCallbacks.BurstFetchMethods = NULL;
+	g_BurstDebugCallbacks.BurstFetchFiles = NULL;
+	g_BurstDebugCallbacks.BurstFetchDebugInfo = NULL;
+	g_BurstDebugCallbacks.BurstFetchMethodInfo = NULL;
+	g_BurstDebugCallbacks.BurstFetchParamInfo = NULL;
+	g_BurstDebugCallbacks.BurstFetchMethodName = NULL;
+	burst_unlock();
+}
+
+// Called from the main thread - once only
+static void burst_mono_install_hooks_imp(BurstMonoDebuggerCallbacks* callbacks,void* domainInitCallback)
+{
+	mono_coop_mutex_init_recursive(&g_BurstDebugMutex);
+	g_BurstDebugMutexInitialised = TRUE;
+
+	burst_lock();
+	// Copy the callback structure (the one passed in may not persist)
+	g_BurstDebugCallbacks.BurstFinishSetup = callbacks->BurstFinishSetup;
+	g_BurstDebugCallbacks.BurstIsFileBursted = callbacks->BurstIsFileBursted;
+	g_BurstDebugCallbacks.BurstBreakpointNotify = callbacks->BurstBreakpointNotify;
+	g_BurstDebugCallbacks.BurstFetchMethods = callbacks->BurstFetchMethods;
+	g_BurstDebugCallbacks.BurstFetchFiles = callbacks->BurstFetchFiles;
+	g_BurstDebugCallbacks.BurstFetchDebugInfo = callbacks->BurstFetchDebugInfo;
+	g_BurstDebugCallbacks.BurstFetchMethodInfo = callbacks->BurstFetchMethodInfo;
+	g_BurstDebugCallbacks.BurstFetchParamInfo = callbacks->BurstFetchParamInfo;
+	g_BurstDebugCallbacks.BurstFetchMethodName = callbacks->BurstFetchMethodName;
+
+	// Provide the callee with access to the features it will need
+	callbacks->buffer_add_byte = buffer_add_byte;
+	callbacks->buffer_add_int = buffer_add_int;
+	callbacks->buffer_add_id = buffer_add_id;
+	callbacks->buffer_add_string = buffer_add_string;
+	callbacks->buffer_add_ptr_id = buffer_add_ptr_id;
+	callbacks->mono_burst_shutdown = burst_mono_shutdown;
+	callbacks->mono_burst_type_load = domainInitCallback;
```

Comment:
Why does `domainInitCallback` get passed through and stored here? I don't see anything on the mono side access it?

Reply:
It was to save unity having to have a copy of the callback structure.

Reply:
ill rename it so its clear its calling into burst.cpp not mono.